tools/mvn/src/main/java/org/wildfly/extras/transformer/tool/maven/SourceTransformerMojo.java

```diff
@@ -231,4 +243,44 @@ private File getOutputDirectory(LifecyclePhase phase) throws IOException {
         }
         return outputDirectory.toFile();
     }
+
+    private Set<String> getIgnored(LifecyclePhase phase) throws IOException {
+        if (!ignoreExisting) {
+            return Collections.emptySet();
+        }
+        switch (phase) {
+            case GENERATE_TEST_SOURCES:
+                return getSourceFiles(Paths.get(mavenProject.getBuild().getTestSourceDirectory()));
+            case GENERATE_RESOURCES:
+                return getSourceFiles(mavenProject.getResources());
+            case GENERATE_TEST_RESOURCES:
+                return getSourceFiles(mavenProject.getTestResources());
+            case GENERATE_SOURCES:
+            default:
```

Comment:
I think default: should either fail or return Collections.emptySet(). If you look at both of the main code paths in execute() (inputFile.isDirectory() and otherwise) they only work properly for the 4 listed phases, although the inputFile.isDirectory() branch isn't clean about that. So we shouldn't increase the existing code ambiguity about what phases are supported.

Returning Collections.emptySet() gets you out of the mode of having to deal with any problems in the (probably unused) inputFile.isDirectory() branch. But that should probably have a "// TODO fail here?" comment.

Reply:
Ah, good point. I'll change that now.